server/src/server_config.rs

```diff
@@ -179,3 +186,62 @@ impl Default for ChaosParams {
         }
     }
 }
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct BucketParams {
+    #[serde(default = "default_bucket_name")]
+    pub bucket_name: String,
+
+    #[serde(default = "default_bucket_region")]
+    pub bucket_region: String,
+
+    #[serde(default)]
+    pub bucket_access_key: String,
+
+    #[serde(default)]
+    pub bucket_secret_key: String,
+
+    #[serde(default)]
+    pub data_path_key: String,
+
+    #[serde(default = "default_bucket_timeout_in_ms")]
+    pub bucket_timeout_in_ms: u32,
+}
+
+impl Default for BucketParams {
+    fn default() -> Self {
+        BucketParams {
+            bucket_name: default_bucket_name(),
+            bucket_region: default_bucket_region(),
+            bucket_access_key: "".to_string(),
+            bucket_secret_key: "".to_string(),
+            data_path_key: "".to_string(),
+            bucket_timeout_in_ms: default_bucket_timeout_in_ms(),
+        }
+    }
+}
+
+pub fn default_bucket_name() -> String {
+    "loki".to_string()
+}
+
+pub fn default_bucket_region() -> String {
+    "eu-west-1".to_string()
+}
+
+pub fn default_bucket_timeout_in_ms() -> u32 {
+    30_000
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum StorageType {
```

Comment:
```suggestion
pub enum DataSource{
```
since this is not really related to "storage" but more "where does my data comes from"